misc/python/materialize/cli/mzcompose.py

```diff
@@ -243,6 +265,10 @@ def parse_known_args(
                 raise errors.BadSpec(
                     f'unknown build mode {pargs.mz_build_mode!r} (expected "dev" or "release")'
                 )
+            if "--mz-build-mode" in pargs.remainder:
+                raise errors.BadSpec(
+                    f"--mz-build-mode must be provided before the command: '{pargs.command}'"
+                )
```

Comment:
This seems like a bug that we should fix.

Reply:
Agreed, but at least giving an error message is better than the current state where it's completely ignored.